File: Lesson 10 Prime and composite numbers/CountFactors.c
#include <math.h>

int solution(int N) {
    
    int count = 0;
    int i = 1;

    while(pow(i, 2) < N)    // i*i will overflow
    {
        if(N%i == 0)
            {count+=2;}
        i += 1;
    }

    if(pow(i, 2) == N)
        {count += 1;}
	
    return count;
}

File: Lesson 10 Prime and composite numbers/CountFactors.py
import math 

def solution(N):
    count = 0
    i = 1
    while(i*i < N):
        if (N%i) == 0:
            count += 2
        i += 1

    if (i*i == N):
        count += 1
    
    return count


def solution(N):
    count = 0
    i = 0
    for i in range(1, math.ceil(math.sqrt(N))):
        if (N%i) == 0:
            count += 2
    
    i += 1
    if (i*i == N):
        count += 1
    
    return count